app/src/main/java/com/example/evently/ui/auth/AuthActivity.java

```diff
@@ -97,19 +101,84 @@ private void showRegisterForm() {
         getSupportFragmentManager()
                 .setFragmentResultListener(
                         RegisterFragment.resultKey, this, (var key, var bundle) -> {
-                            // TODO (chase): Persist the data in the bundle to register user.
-                            // NOTE: If the bundle contains an email that already exists in the DB,
-                            // show an error and prompt the user to login instead of registering.
-                            startActivity(transition);
-                            finish();
+                            String email = bundle.getString("email");
+                            String name = bundle.getString("name");
+                            String phone = bundle.getString("phone");
+
+                            // Stores a new account
+                            AccountDB accountDB = new AccountDB();
+                            accountDB.fetchAccount(
+                                    email,
+                                    optionalAccount -> {
+                                        if (optionalAccount.isEmpty()) {
+                                            hasRegisterForm = true;
```

Comment:
This shouldn't be in this flow. It's already set to true regardless but since this flow is about to switch to another activity - it doesn't matter.